kinesis_consumer.cpp

```diff
@@ -148,6 +149,28 @@ lookup_region(const char *r)
 	return -1;
 }
 
+static void
+parse_url(ClientConfiguration &config,
```

Comment:
I can't see any hard failure here. As written it is meant to pass on the constituent parts to the next module that uses them.

Are you referring to that it accepts malformed URLs?